### DISCUSSION TITLE
Cypress Component Testing with Angular

### DISCUSSION THREAD

**Discussion by imdurdydan** [nrwl/nx#11758] [General]
Does nx have a generator or easily support component testing with cypress for angular applications? My angular.json is using version 2 to support multiple angular applications and each application has its own project json. I found this generator and tried to use it `nx g @nrwl/cypress:cypress-component-project --project={project}`. It generates everything just fine but when I try to run any tests or even `npx cypress open` it gives me this error:

`Could not find a project with projectType "application" in "angular.json`. It seems like cypress component testing does not play nicely with the version 2 of angular.json schema or im doing something incredibly wrong :) (probably the second). 

Any insight would be appreciated!

My angular.json:
`{
  "$schema": "./node_modules/nx/schemas/workspace-schema.json",
  "version": 2,
  "projects": {
    "accelerator-engage": "apps/accelerator-engage",
    "accelerator-engage-e2e": "apps/accelerator-engage-e2e",
    "accelerator-message": "apps/accelerator-message",
    "accelerator-message-e2e": "apps/accelerator-message-e2e",
    "accelerator-segment": "apps/accelerator-segment",
    "accelerator-segment-e2e": "apps/accelerator-segment-e2e",
    "accelerator-shell": "apps/accelerator-shell",
    "accelerator-shell-e2e": "apps/accelerator-shell-e2e",
    "cross-product-services": "libs/cross-product-services",
    "cross-product-types": "libs/cross-product-types",
    "dashboard": "apps/dashboard",
    "dashboard-e2e": "apps/dashboard-e2e",
    "design-system": "libs/design-system",
    "design-system-e2e": "apps/design-system-e2e"
  }
}
`

Side note: I do see generators for this for react but not angular

**Comment by barbados-clemens** [comment]
tracked via https://github.com/nrwl/nx/issues/11602
